src/Definition/Doc.elm

```diff
@@ -612,104 +672,127 @@ decodeSpecialForm =
         ]
 
 
-decode : Decode.Decoder Doc
-decode =
+decode_ : TreePath -> Decode.Decoder Doc
+decode_ path =
```

Comment:
Primary change is to collect paths via TreePath for giving `Folded` and `Source` ids to fold on.